commands/fetch_templates.go

```diff
@@ -4,263 +4,171 @@
+func canWriteLanguage(availableLanguages map[string]bool, language string, overwrite bool) bool {
+	canWrite := false
+	if availableLanguages != nil && len(language) > 0 {
+		if _, found := availableLanguages[language]; found {
+			return availableLanguages[language]
+		}
+		canWrite = templateFolderExists(language, overwrite)
+		availableLanguages[language] = canWrite
+	}
 
-	zipFile, err := zip.OpenReader(archivePath)
+	return canWrite
+}
 
-	if err != nil {
-		return nil, nil, err
+// Takes a language input (e.g. "node"), tells whether or not it is OK to download
+func templateFolderExists(language string, overwrite bool) bool {
+	dir := templateDirectory + language
+	if _, err := os.Stat(dir); err == nil && !overwrite {
+		// The directory template/language/ exists
+		return false
 	}
-	defer zipFile.Close()
-
-	for _, z := range zipFile.File {
+	return true
+}
 
-		relativePath := z.Name[strings.Index(z.Name, "/")+1:]
-		if strings.Index(relativePath, "template/") != 0 {
-			// Process only directories inside "template" at root
-			continue
-		}
+func moveTemplates(repoPath string, overwrite bool) ([]string, []string, error) {
+	var (
+		existingLanguages []string
+		fetchedLanguages  []string
+		err               error
+	)
 
-		action, language, isDirectory := canExpandTemplateData(availableLanguages, relativePath)
+	availableLanguages := make(map[string]bool)
 
-		var expandFromZip bool
+	templateDir := filepath.Join(repoPath, templateDirectory)
+	templates, err := ioutil.ReadDir(templateDir)
+	if err != nil {
+		return nil, nil, fmt.Errorf("can't find templates in: %s", repoPath)
+	}
 
-		switch action {
+TEMPLATES:
+	for _, file := range templates {
+		if !file.IsDir() {
+			continue TEMPLATES
+		}
+		language := file.Name()
 
-		case shouldExtractData:
-			expandFromZip = true
-		case newTemplateFound:
-			expandFromZip = true
+		canWrite := canWriteLanguage(availableLanguages, language, overwrite)
+		if canWrite {
 			fetchedLanguages = append(fetchedLanguages, language)
-		case directoryAlreadyExists:
-			expandFromZip = false
+			// Do cp here
+			languageSrc := filepath.Join(templateDir, language)
+			languageDest := filepath.Join(templateDirectory, language)
+			copy(languageSrc, languageDest, file)
+		} else {
 			existingLanguages = append(existingLanguages, language)
-		case skipWritingData:
-			expandFromZip = false
-		default:
-			return nil, nil, fmt.Errorf(fmt.Sprintf("don't know what to do when extracting zip: %s", archivePath))
-		}
-
-		if expandFromZip {
-			var rc io.ReadCloser
-
-			if rc, err = z.Open(); err != nil {
-				break
-			}
-			defer rc.Close()
-
-			if err = createPath(relativePath, z.Mode()); err != nil {
-				break
-			}
-
-			// If relativePath is just a directory, then skip expanding it.
-			if len(relativePath) > 1 && !isDirectory {
-				if err = writeFile(rc, z.UncompressedSize64, relativePath, z.Mode()); err != nil {
-					return nil, nil, err
-				}
-			}
+			continue TEMPLATES
```

Comment:
Can we avoid using goto / `continue`?

Reply:
Just removed the "goto".  Do you also mean to completely remove the "continue" as well?